pwndbg/commands/got_tracking.py

```diff
@@ -0,0 +1,237 @@
+
+
+def got_report(soname=".*", writable=False, fnname=".*"):
+    """
+    Prints out a report of the current status of the GOT tracker.
+    """
+    if not pwndbg.gdblib.got.GOT_TRACKING:
+        print(message.error("GOT call tracking is not enabled"))
+        return
+
+    soname = re.compile(soname)
+    fnname = re.compile(fnname)
+
+    print(
+        f"Showing {'writable' if writable else 'all'} GOT function entries and how many times they were called."
+    )
+    print()
+
+    per_object = {}
+    for _, (tracker, patcher) in pwndbg.gdblib.got.all_tracked_entries():
+        objname = tracker.link_map_entry.name()
+        if objname == b"":
+            objname = pwndbg.gdblib.proc.exe
+        else:
+            objname = pwndbg.gdblib.got.display_name(objname)
+
+        # Filter out objects we're not interested in.
+        if soname.match(objname) is None:
+            continue
+
+        if objname not in per_object:
+            per_object[objname] = []
+        per_object[objname].append((tracker, patcher))
+
+    for objname, trackers in per_object.items():
+        print(f"Calls from {objname}:")
+        rows = [["Address in GOT", "Function Address", "Symbol", "Call Count"]]
+        for tracker, patcher in trackers:
+            # If requested, filter out entries that are not in a writable
+            # portion of memory.
+            if writable and not pwndbg.gdblib.vmmap.find(patcher.entry).write:
+                continue
+
+            dynamic = tracker.dynamic_section
+            sym_index = tracker.relocation_fn(tracker.relocation_index, "r_sym")
+            sym_name = dynamic.symtab_read(sym_index, "st_name")
+            sym_name = pwndbg.gdblib.got.display_name(dynamic.string(sym_name))
+
+            # Filter out symbols we're not interested in.
+            if fnname.match(sym_name) is None:
+                continue
+
+            if sym_name == "":
+                sym_name = "<Empty>"
+
+            hits = tracker.total_hits
+            hits = f"{hits} hit{'s' if hits != 1 else ''}"
+            rows.append([f"{tracker.target:#x}", f"{patcher.entry:#x}", sym_name, hits])
+        columns(rows)
+        print()
+
+
+def got_tracking_status(address):
+    """
+    Prints out information about a single GOT tracking entry.
+    """
+    if not pwndbg.gdblib.got.GOT_TRACKING:
+        print(message.error("GOT call tracking is not enabled"))
+        return
+
+    try:
+        address = int(address, 0)
+    except ValueError as e:
+        print(message.error(f"Invalid address {address}: {e}"))
+        return
+
+    result = pwndbg.gdblib.got.tracked_entry_by_address(address)
+    if result is None:
+        print(message.error(f"No entry at address {address:#x}"))
+        print("Hint: This command expects the address of the entry in the GOT. So, consider")
+        print("using the address from the 'Address in GOT' column of the `track-got info`")
```

Comment:
```suggestion
        print("using the address from the 'Function Address' column of the `track-got info`")
```

```
Calls from /lib64/ld-linux-x86-64.so.2:
Address in GOT Function Address Symbol               Call Count
0x7ffff7ed0820 0x7ffff7ffd018   _dl_catch_exception  0 hits
0x7ffff7ed0770 0x7ffff7ffd020   _dl_signal_exception 0 hits
0x7ffff7ed07c0 0x7ffff7ffd028   _dl_signal_error     0 hits
0x7ffff7ed0940 0x7ffff7ffd030   _dl_catch_error      0 hits

pwndbg> track-got query 0x7ffff7ed0820
No entry at address 0x7ffff7ed0820
Hint: This command expects the address of the entry in the GOT. So, consider
using the address from the 'Address in GOT' column of the `track-got info`
command.
pwndbg> track-got query 0x7ffff7ffd018
Tracking details for _dl_catch_exception

Dynamic object name: /lib64/ld-linux-x86-64.so.2
Jump target address: 0x7ffff7ed0820
GOT entry address:   0x7ffff7ffd018
Relocation index:    0
Symbol index:        17
```

Seems this must be 'Function Address' column

Reply:
Or are the two mixed up?

Reply:
They're backwards in `track-got info`. It should be `Function Address Address in GOT Symbol               Call Count`, or the numbers should be the other way round. I must've gotten them mixed up when I changed the way these were displayed.